src/mongoc/mongoc-topology.c

```diff
@@ -351,29 +359,23 @@ mongoc_topology_select (mongoc_topology_t         *topology,
          topology->stale = false;
       }
 
-      /* until we find a server or timeout */
-      for (;;) {
```

Comment:
Nice fix. mongoc_topology_description_select already handles looping until the timeout, right?